ibmcloud/README.md

```diff
@@ -0,0 +1,207 @@
+
+The setup procedure includes the following sub tasks.
+
+* Create a Virtual Private Cloud (VPC) including security groups, subnet, and gateway
+* Create a Kubernetes cluster on two virtual server instances (VSIs)
+* Build a custom VM image for pod VMs
+* Install cloud-api-adaptor on a worker node
+* Run a demo
+
+## Prerequisites
+
+To automate preparation of VPC and VSIs, you need to install terraform and ansible on your client machine. Please follow the the official installation guides.
+
+* [Install Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+* [Install Ansible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html)
+
+Optionally, you can install IBM Cloud CLI.
+
+* [Installing the stand-alone IBM Cloud CLI](https://cloud.ibm.com/docs/cli?topic=cli-install-ibmcloud-cli)
+
+Note that you can use the IBM Cloud Web UI for most of the operations of IBM Cloud.
+
+* [https://cloud.ibm.com/vpc-ext/overview](https://cloud.ibm.com/vpc-ext/overview)
+
+You need IBM Cloud API key. You can create your own API key at [https://cloud.ibm.com/iam/apikeys](https://cloud.ibm.com/iam/apikeys).
+
+## Create a VPC
+
+First, you need to create a Virtual Private Cloud (VPC). The Terraform configuration files are in [ibmcloud/terraform/common](./terraform/common/).
+
+To use the Terraform configuration, you need to create a file `terraform.tfvars` at in the same directory of the other files of the Terraform configuration to specify your IBM Cloud API Key. `terraform.tfvars` looks like this.
+```
+ibmcloud_api_key = "<your API key>"
+```
+You can also customize the other parameters by specifying custom values in `terraform.tfvars`. The default values of such parameters are defined in [variables.tf](./terraform/common/variables.tf)
+
+Then, you can create your VPC by executing the following commands.
+
+```
+cd ibmcloud/terraform/common
+terraform init
+terraform plan
+terraform apply
+```
+
+The following cloud resources will be created. Please check [main.tf](terraform/common/main.tf) for the details.
+* VPC
+* Security groups
+* Subnets
+* Public gateway
+* Floating IP for the public gateway
+
+## Create a Kubernetes cluster
+
+Another Terraform configuration is available at [ibmcloud/terraform/cluster](./terraform/cluster) to create a
+Kubernetes cluster on the VPC you just created. Note that you can create multiple clusters by using different cluster names.
+
+As usual, you need to create `terraform.tfvars` to specify parameter values. `terraform.tfvars` looks like this.
+
+```
+ibmcloud_api_key = ibmcloud_api_key = "<your API key>"
+ssh_key_name = "<your SSH key name>"
+cluster_name = "<cluster name>"
+```
+
+`ssh_key_name` is a name of your SSH key registered in IBM Cloud.
+You can add your SSH key at [https://cloud.ibm.com/vpc-ext/compute/sshKeys](https://cloud.ibm.com/vpc-ext/compute/sshKeys). This ssh key will be installed on control-plane and worker nodes.
```

Comment:
It might be worth specifying here to create the ssh-key in the Tokyo region through the web UI (or updating the region in terraform variables I guess). I defaulted to Frankfurt when I opened the link which mean when I ran
```
terraform plan
```
I hit a problem:
```
╷
│ Error: No SSH Key found with name peer-pod-fyre
│
│   with data.ibm_is_ssh_key.ssh_key,
│   on main.tf line 16, in data "ibm_is_ssh_key" "ssh_key":
│   16: data "ibm_is_ssh_key" "ssh_key" {
│
```
so eventually I worked out that is was in the wrong region and that fixed the issue.

Reply:
I added description about region